lib/calculate.action.js

```diff
@@ -22,16 +25,17 @@ module.exports = async (ticker) => {
   action.value = multiply(action.currentHolding, action.exchangeRate);
   // make sure our target can handle selling the vol
   // we want our position to grow by the expected gain + our vol target + the last target (accumulates APY expectation)
-  action.target = add(config.vol, action.expectedGain, memory.lastLog.Target);
+  action.target = add(vol, action.expectedGain, memory.lastLog.Target);
   if (memory.lastLog.Funds < 0) {
     // however, if we sold on the last action, that vol shouldn't add to our growth target
     action.target = add(action.target, memory.lastLog.Funds);
   }
   action.diff = subtract(action.value, action.target);
   // simply, if our account value is above the target, sell, else buy
   // the `funds` is our market buy/sell amount in the base currency
-  const tradeValue = process.env.CPBB_LIQUIDATE ? action.value : config.vol;
-  action.funds = action.diff > 0 ? tradeValue * -1 : config.vol;
+  const tradeValue = process.env.CPBB_LIQUIDATE ? action.value : vol;
+  // in the case of a limit order that was filled, we are forcing the calculation to have already bought (can't calc a sell)
+  action.funds = forceBuy ? tradeValue : (action.diff > 0 ? tradeValue * -1 : tradeValue);
```

Comment:
wondering if we might add a forceSell in the future. This whole file could use some architectural cleanup. I'll think about it more later.